assets/src/onboarding-wizard/components/navigation-context-provider.js

```diff
@@ -24,13 +24,25 @@ export const Navigation = createContext();
 export function NavigationContextProvider( { children, pages } ) {
 	const [ activePageIndex, setActivePageIndex ] = useState( 0 );
 	const [ canGoForward, setCanGoForward ] = useState( true ); // Allow immediately moving forward on first page. @todo This may need to change in 2.1.
-	const { editedOptions } = useContext( Options );
+	const { editedOptions, readerModeWasOverridden } = useContext( Options );
 
 	const { theme_support: themeSupport } = editedOptions;
 
-	const currentPage = pages[ activePageIndex ];
+	const adaptedPages = useMemo( () => {
+		if ( 'reader' === themeSupport ) {
+			return pages;
+		}
+
+		if ( readerModeWasOverridden ) {
+			setActivePageIndex( ( index ) => index - 1 );
+		}
```

Comment:
I'm not sure if this is the right place to do this mutation. @johnwatkins0 any recommendations?

Reply:
Yeah, setting state from inside `useMemo` doesn't seem quite right. How about breaking this out into an effect hook?

```
useEffect( () => {
	if ( readerModeWasOverridden ) {
		setActivePageIndex( ( index ) => index - 1 );
	}
}, [ readerModeWasOverridden ] );
```

If you do something like this, then the `useMemo` hook might not even be needed anymore; you could do a `READER === themeSupport ? pages : pages.filter( ( page ) => 'theme-selection' !== page.slug )` in the Navigation.Provider value object.

Reply:
After some testing I found out If I set `pages` directly in `Navigation.Provider `, it requires much more additional logic to keep `activePageIndex` in sync with `pages`, including having to modify `moveBack()` and `moveForward()`. Also, memoizing the adapted `pages` array should give us a small performance gain, if any.

In https://github.com/ampproject/amp-wp/pull/5123/commits/28e57cc7a2eb5fed518324f6edfc10d98cc97c0f I'm using `useEffect` to update `activePageIndex` as suggested.